libs/ui/src/modals/screens/slippage-settings/slippage-settings.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { FC, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { View } from 'react-native';
+import { useDispatch } from 'react-redux';
+
+import { Warning } from '../../../components/announcement/components/warning/warning';
+import { SpeedSelector } from '../../../components/speed-selector/speed-selector';
+import { Text } from '../../../components/text/text';
+import { TextInput } from '../../../components/text-input/text-input';
+import { ScreensEnum } from '../../../enums/sreens.enum';
+import { useNavigation } from '../../../hooks/use-navigation.hook';
+import { setSlippageToleranceAction } from '../../../store/settings/settings.actions';
+import { useSlippageToleranceSelector } from '../../../store/settings/settings.selectors';
+import { ModalActionContainer } from '../../components/modal-action-container/modal-action-container';
+
+import { slippageOptions, ownSlippageRules, OWN } from './constatns';
+import { styles } from './slippage-settings.styles';
+import { findSlippageOption } from './utils/find-slippage-option.util';
+
+export const SlippageSettings: FC = () => {
+  const dispatch = useDispatch();
+  const { goBack, navigate } = useNavigation();
+  const slippageTolerance = useSlippageToleranceSelector();
+
+  const [slippage, setSlippage] = useState(findSlippageOption(slippageTolerance));
+
+  const {
+    control,
+    formState: { errors },
+    handleSubmit,
+    clearErrors,
+    watch
+  } = useForm<{ ownSlippage: string }>({
+    mode: 'onChange',
+    defaultValues: {
+      ownSlippage: slippageTolerance
+    }
+  });
+  const ownSlippageTolerance = watch('ownSlippage');
+
+  const isOwnSlippageSelected = slippage.title === OWN;
+  const showWarning = isOwnSlippageSelected && Number(ownSlippageTolerance) >= 4 && Number(ownSlippageTolerance) <= 50;
+  const isSubmitDisabled = Object.keys(errors).length > 0;
+
+  const onSelect = (option: { title: string; value: string }) => {
+    setSlippage(option);
+
+    if (option.value !== OWN) {
+      clearErrors();
+    }
+  };
+
+  const onSubmit = ({ ownSlippage }: { ownSlippage: string }) => {
+    dispatch(setSlippageToleranceAction(isOwnSlippageSelected ? Number(ownSlippage).toString() : slippage.value));
```

Comment:
![image](https://user-images.githubusercontent.com/35599200/214325228-5525c89b-4688-4dc7-9a4c-2f35c8ef2cbc.png)
because user can save such kind of numbers